### PR TITLE
chore: rename NsPubsubTopic

### PR DESCRIPTION
**Pull request by gabrielmer**
# Description

Now that only static sharding is supported and we want to use a type for clusterId + shardId, we're renaming the `NsPubsubTopic` type to `RelayShard` for it to have a more meaningful name and use it throughout the codebase.

# Changes

- [x] renamed `NsPubsubTopic` to `RelayShard`

## Issue

Advances #2806

### CODE REVIEW

File: waku/waku_core/topics/pubsub_topic.nim [hunk 1]
```diff
@@ -67,19 +67,19 @@ proc parseStaticSharding*(
...
 # Pubsub topic compatibility
 
-converter toPubsubTopic*(topic: NsPubsubTopic): PubsubTopic =
+converter toPubsubTopic*(topic: RelayShard): PubsubTopic =
   $topic
 
-proc `==`*[T: NsPubsubTopic](x, y: T): bool =
+proc `==`*[T: RelayShard](x, y: T): bool =
   if x.clusterId != y.clusterId:
     return false
```
Comment (SionoiS) [inline waku/waku_core/topics/pubsub_topic.nim:84]: That would only be partial equality. What happen is shard don't match?

**Review by SionoiS — CHANGES_REQUESTED** [review]
LGTM expect the equality bit.

File: waku/waku_core/topics/pubsub_topic.nim (hunk 1 above)
Comment (gabrielmer) [reply waku/waku_core/topics/pubsub_topic.nim:84]: I think the other part of the proc was not properly displayed

image


I think it's ok like that right?

Comment (SionoiS) [reply waku/waku_core/topics/pubsub_topic.nim:84]: Ah! NVM then.!

**Review by SionoiS — APPROVED** [review]

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100:

**Pull request merged**